src/nih_wayfinding/app/scripts/views/navigate/navigate-controller.js

```diff
@@ -38,6 +45,130 @@
 
             // Subscribe to the location update event
             $scope.$on('nih.navigation.positionUpdated', onPositionUpdated);
+            $scope.$on('$stateChangeStart', Navigation.stopIntervalTask);
+        }
+
+        function setDefaultFooter() {
+            ctl.footer = {
+                left:  {
+                    text: 'Reroute',
+                    func: function() { $state.go('reroute'); }
```

Comment:
might be helpful to change this from 'func' to 'onClick'. It's a bit more descriptive and follows relatively standardized naming conventions for event handler methods.